File: 1422 maximum-score-after-splitting-a-string/solution.go
func maxScore(s string) int {
    o,z,m:=0,0,0
    for _,v :=range s {
        if v=='1' {
            o++
        }
    }
    for i:=0;i<len(s)-1;i++ {
        if s[i]=='0' {
            z++
        } else {
            o--
        }
        m=max(m,o+z)
        }
        return m
}

File: 1422 maximum-score-after-splitting-a-string/solution.cpp
class Solution {
public:
    int maxScore(string s) {
        int o = 0,z = 0,ans = INT_MIN;
        for(char c:s) if(c-'0'==1) o++;
        for(int i=0;i<s.size()-1;i++) {
            int v = s[i]-'0';
            if(v==0) z++;
            else o--;
            ans = max(z+o,ans);
        }
        return ans;
    }
};
